src/client/index.ts

```diff
@@ -0,0 +1,55 @@
+import { WalletProviderResponseError } from "../utils/errors";
+import {
+  ProblemDetail,
+  createApiClient as createWalletProviderApiClient,
+} from "./generated/wallet-provider";
+import { ApiClient as WalletProviderApiClient } from "./generated/wallet-provider";
+
+export type WalletProviderClient = WalletProviderApiClient;
+
+const validateResponse = async (response: Response) => {
+  if (!response.ok) {
+    var problemDetail: ProblemDetail = {};
+    try {
+      problemDetail = ProblemDetail.parse(await response.json());
+    } catch (_) {
+      problemDetail = {
+        title: "Invalid response from Wallet Provider",
+        detail: "Unable to parse response to ProblemDetail",
+      };
+    }
+
+    throw new WalletProviderResponseError(
+      problemDetail.title
+        ? problemDetail.title
+        : "Invalid response from Wallet Provider",
+      problemDetail.type,
+      problemDetail.detail
+    );
+  }
+  return response;
+};
+
+export function getWalletProviderClient(context: {
+  walletProviderBaseUrl: string;
+  appFetch?: GlobalFetch["fetch"];
+}) {
+  const { walletProviderBaseUrl, appFetch = fetch } = context;
+
+  return createWalletProviderApiClient(
```

Comment:
shouldn't we handle the case where the `fetch` fails?

Reply:
True, I hadn't thought of that. Then I would say not to catch

Reply:
We might refine this by exposing our own subset of errors including the error code. This could be especially useful to distinguish `401` returned from our backend or from our partners which might have a different meaning (implying that there are flows with mixed calls). That's just an idea though, it might unwrap differently later on.

Reply:
The error codes, however, do not imply a fetch throw but simply a response with a different status code. For now the `validateResponse` in case of error throws a classic `WalletProviderResponseError` with the details of the error. Maybe with a future PR this could be specialized. 
For now I would say to leave it like this since the error is telling anyway.

Reply:
I added more details about the status code error on `validateResponse`